moped-editor/src/views/projects/projectView/ProjectComponents/CreateComponentDrawTools.js

```diff
@@ -11,13 +12,26 @@ import mapboxDrawStylesOverrides from "src/styles/mapboxDrawStylesOverrides";
  * @param {Function} setIsDrawing - function to update if we are drawing or not
  * @returns {JSX.Element}
  */
+
 const CreateComponentDrawTools = ({
   createDispatch,
   linkMode,
   setCursor,
   setIsDrawing,
 }) => {
   const drawControlsRef = useRef();
+  const [isTrashButtonClickable, toggleTrashButtonClickable] = useState(false);
+  React.useEffect(() => {
```

Comment:
Since this combination of state and effect needs to go in the create and edit draw tool components, it would be slick to turn this into a custom hook! I don't think there are anything that would prevent doing that, but I could have overlooked something.